### PR TITLE
Fix a bug in UpgradeAddonsBeforeControlPlane feature gate e2e test

### PR DESCRIPTION
**Pull request by SataQiu**
Ref #2884

### CODE REVIEW

File: kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func upgradeKubeadmBinary(n *status.Node, upgradeVersion *K8sVersion.Version) er
 	return nil
 }
 
-func kubeadmUpgradeApply(c *status.Cluster, cp1 *status.Node, upgradeVersion *K8sVersion.Version, patchesDir string, featureGates string, wait time.Duration, vLevel int) error {
+func kubeadmUpgradeApply(c *status.Cluster, cp1 *status.Node, upgradeVersion *K8sVersion.Version, patchesDir string, featureGate string, wait time.Duration, vLevel int) error {
```
Comment (pacoxu) [inline kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go:123]: Overall LGTM.

why change featureGates to featureGate? It is confusing.

Comment (SataQiu) [reply kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go:123]: We can only support a single kubeadm feature-gate now.

Comment (SataQiu) [reply kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go:123]: // FeatureGate option sets **a single kubeadm feature-gate** for the kubeadm commands

Comment (SataQiu) [reply kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go:123]: For example: https://github.com/kubernetes/kubeadm/blob/2f7c488859fb8c75b4037903dc7895bf05ccec1c/kinder/pkg/cluster/manager/actions/kubeadm-config.go#L97-L100

Comment (pacoxu) [reply kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go:123]: Sorry, It seems to be limited at the beginning.

https://github.com/kubernetes/kubeadm/blob/2f7c488859fb8c75b4037903dc7895bf05ccec1c/kinder/cmd/kinder/do/do.go#L126-L130

**Pull request merged**